test/gateway/eip20_cogateway/progress_revert_redeem.js

```diff
@@ -0,0 +1,385 @@
+      MessageStatusEnum.DeclaredRevocation,
+    );
+
+  });
+
+  it('should progress revert redeem', async function () {
+
+    await cogateway.setStorageRoot(revertRedeemParams.blockHeight, revertRedeemParams.storageRoot);
+
+    let tx = await cogateway.progressRevertRedeem(
+      revertRedeemParams.messageHash,
+      revertRedeemParams.blockHeight,
+      revertRedeemParams.rlpParentNodes,
+    );
+
+    assert.equal(
+      tx.receipt.status,
+      1,
+      "Receipt status is unsuccessful",
+    );
+
+  });
+
+  it('should emit `RedeemReverted` event ', async function () {
+
+    await cogateway.setStorageRoot(revertRedeemParams.blockHeight, revertRedeemParams.storageRoot);
+
+    let tx = await cogateway.progressRevertRedeem(
+      revertRedeemParams.messageHash,
+      revertRedeemParams.blockHeight,
+      revertRedeemParams.rlpParentNodes,
+    );
+
+    let event = EventDecoder.getEvents(tx, cogateway);
+    let eventData = event.RedeemReverted;
+
+    assert.isDefined(
+      event.RedeemReverted,
+      'Event `RedeemReverted` must be emitted.',
+    );
+
+    assert.strictEqual(
+      eventData._messageHash,
+      revertRedeemParams.messageHash,
+      `Message hash from the event must be equal to ${revertRedeemParams.messageHash}.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemer,
+      revertRedeemParams.redeemer,
+      `Redeemer address from the event must be equal to ${revertRedeemParams.redeemer}.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemerNonce.eq(revertRedeemParams.nonce),
+      true,
+      `Redeemer nonce from the event must be equal to ${revertRedeemParams.nonce}.`,
+    );
+
+    assert.strictEqual(
+      eventData._amount.eq(revertRedeemParams.amount),
+      true,
+      `Redeem amount from the event must be equal to ${revertRedeemParams.amount}.`,
+    );
+
+  });
+
+  it('should burn bounty and penalty amount ', async function () {
+
+    let burnerAddress = await cogateway.burner.call();
+
+    let cogatewayInitialTokenBalance = await Utils.getBalance(cogateway.address);
+    let redeemerInitialTokenBalance = await Utils.getBalance(revertRedeemParams.redeemer);
+    let burnerInitialTokenBalance = await Utils.getBalance(burnerAddress);
+
+    await cogateway.setStorageRoot(revertRedeemParams.blockHeight, revertRedeemParams.storageRoot);
+
+    await cogateway.progressRevertRedeem(
+      revertRedeemParams.messageHash,
+      revertRedeemParams.blockHeight,
+      revertRedeemParams.rlpParentNodes,
+    );
+
+    let cogatewayFinalTokenBalance = await Utils.getBalance(cogateway.address);
+    let redeemerFinalTokenBalance = await Utils.getBalance(revertRedeemParams.redeemer);
+    let burnerFinalTokenBalance = await Utils.getBalance(burnerAddress);
+
+    assert.strictEqual(
+      cogatewayFinalTokenBalance.eq(cogatewayInitialTokenBalance.sub(bountyAmount.muln(2.5))),
+      true,
+      `CoGateway balance must decrease by ${bountyAmount.muln(2.5).toString(10)}.`,
```

Comment:
Why times 2.5? Where does that amount come from (who sent that exact amount to the gateway)?

Reply:
Added comment and extracted variable.